demos/custom-dialogs/custom-dialog.js

```diff
@@ -74,90 +81,91 @@ CustomDialog.hide = function() {
+    // Generate HTML
+    backdropDiv = document.createElement('div');
+    backdropDiv.id = 'customDialogBackdrop';
+    backdropDiv.style.cssText =
+        'position: absolute;' +
+        'top: 0; left: 0; right: 0; bottom: 0;' +
+        'background-color: rgba(0, 0, 0, .7);';
+    document.body.appendChild(backdropDiv);
+
+    dialogDiv = document.createElement('div');
+    dialogDiv.id = 'customDialog';
+    dialogDiv.style.cssText =
+        'background-color: #fff;' +
+        'width: 400px;' +
+        'margin: 20px auto 0;' +
+        'padding: 10px;';
+    backdropDiv.appendChild(dialogDiv);
+
+    dialogDiv.onclick = function(event) {
+      event.stopPropagation();
     };
 
-    var dialogInput = document.getElementById('dialogInput');
-    CustomDialog.inputField = dialogInput;
-    if (dialogInput) {
-        dialogInput.focus();
-
-        dialogInput.onkeyup = function(event) {
-            if (event.keyCode == 13) {
-                // Process as OK when user hits enter.
-                onOkay();
-                return false;
-            } else if (event.keyCode == 27)  {
-                // Process as cancel when user hits esc.
-                onCancel();
-                return false;
-            }
-        };
-    } else {
-        var okay = document.getElementById('dialogOkay');
-        okay && okay.focus();
-    }
-
-    if (options.showOkay) {
-        document.getElementById('dialogOkay').onclick = onOkay;
-    }
-    if (options.showCancel) {
-        document.getElementById('dialogCancel').onclick = onCancel;
-    }
-
-    backdropDiv.onclick = onCancel;
-}
+    CustomDialog.backdropDiv_ = backdropDiv;
+    CustomDialog.dialogDiv_ = dialogDiv;
+  }
+  backdropDiv.style.display = 'block';
+  dialogDiv.style.display = 'block';
+
+  dialogDiv.innerHTML =
+      '<header class="customDialogTitle"></header>' +
+      '<p class="customDialogMessage"></p>' +
+      (options.showInput? '<div><input id="customDialogInput"></div>' : '') +
+      '<div class="customDialogButtons">' +
+      (options.showCancel ? '<button id="customDialogCancel">Cancel</button>': '') +
+      (options.showOkay ? '<button id="customDialogOkay">OK</button>': '') +
+      '</div>';
+  dialogDiv.getElementsByClassName('customDialogTitle')[0]
+      .appendChild(document.createTextNode(title));
+  dialogDiv.getElementsByClassName('customDialogMessage')[0]
+      .appendChild(document.createTextNode(message));
+
+  var onOkay = function(event) {
+    CustomDialog.hide();
+    options.onOkay && options.onOkay();
+    event && event.stopPropagation();
+  };
+  var onCancel = function(event) {
+    CustomDialog.hide();
+    options.onCancel && options.onCancel();
+    event && event.stopPropagation();
+  };
+
+  var dialogInput = document.getElementById('customDialogInput');
+  CustomDialog.inputField = dialogInput;
+  if (dialogInput) {
+    dialogInput.focus();
+
+    dialogInput.onkeyup = function(event) {
+      if (event.keyCode == 13) {
```

Comment:
This is if the button has focus, in which the default button behavior already does this.